pkg/util/template.go

```diff
@@ -214,6 +220,38 @@ func configMapEntriesGenerator(
 	}
 }
 
+// lookup does a dot-separated path lookup on the input struct or map. If the input or any of
+// its children on the targeted path are not a map or struct, it returns nil.
+func lookup(path string, input interface{}) interface{} {
```

Comment:
The `get` function of Sprig receives the input as first argument I believe, which reads with the more naturally as `get <object> <field>`.

There are pros and cons to either ways, taking the input as last argument helps in the pipeline form where the output of the previous command is passed as last argument. I'm a bit concerned that we may trick everyone into getting it wrong if `get` and `lookup` have swapped signatures tho.

How about offering both maybe?

```go
func lookup(input interface{}, path string) interface{} {
  ...
}
```
```go
// naming is hard
func select(path string, input interface{}) interface{} {
  return lookup(input, path)
}
```

Reply:
Oh, interesting, for some reason I thought I was following the `get` convention. Will fix it up.

Reply:
Fixed. For the alternative, named it `pathLookup` since `select` is a golang keyword.